scorecard/scorecard/scorecard.py

```diff
@@ -28,12 +36,97 @@
     'hail': 'hail-is/hail',
 }
 
+asana_tags = {
+    'design-docs': '1194895524341386',
+    'scorecard': '1194895524341389'
+}
+
+asana_to_github_user = {
+    'Jackie Goldstein': 'jigold',
+    'Cotton Seed': 'cseed',
+    'Daniel King': 'danking',
+    'John Compitello': 'johnc1231',
+    'Tim Poterba': 'tpoterba',
+    'Arcturus Wang': 'catoverdrive',
+    'Dania Abuhijleh': 'Dania-Abuhijleh',
+    'Patrick Schultz': 'patrick-schultz'
+}
+
 routes = web.RouteTableDef()
 
 data = None
 timestamp = None
 
 
+class AsanaClient:
+    _session: Optional[ClientSession]
+
+    def __init__(self, *, session: Optional[ClientSession] = None, **kwargs):
+        self._base_url = 'https://app.asana.com/api/1.0'
+        if session is None:
+            session = ClientSession(**kwargs)
+        self._session = session
+
+    async def get(self, path: str, **kwargs) -> Any:
+        async with await self._session.get(
+                f'{self._base_url}{path}', **kwargs) as resp:
+            return await resp.json()
+
+    async def post(self, path: str, **kwargs) -> Any:
+        async with await self._session.post(
+                f'{self._base_url}{path}', **kwargs) as resp:
+            return await resp.json()
+
+    async def delete(self, path: str, **kwargs) -> None:
+        async with await self._session.delete(
+                f'{self._base_url}{path}', **kwargs) as resp:
+            return await resp.json()
+
+    async def close(self) -> None:
+        if self._session is not None:
+            await self._session.close()
+            self._session = None
+
+    async def __aenter__(self: ClientType) -> ClientType:
```

Comment:
Usually `self` doesn't have a type, right? I think the return type can be `'AsanaClient'`, no?

Reply:
This is based on the aiogoogle client Cotton wrote.